src/Client.php

```diff
@@ -25,6 +25,12 @@ class Client
      */
     private $emailFactory;
 
+      /**
+     * 
+     * @var boolean
+     */
+    private $peek;
```

Comment:
You should not store state into a property in this class. The `Client` class is a service, it should be stateless.

You should instead pass the `$peek` value in parameters of method calls.